advanced_bank_reconciliation/advanced_bank_reconciliation/doctype/advance_bank_reconciliation_tool/tests/test_partial_allocation.py

```
@@ -489,3 +492,692 @@ def test_normalise_helper_cascade_negative_credit_note(self):
+		This isolates the update_allocated_amount formula from the PE creation
+		and link-validation layers. rows: list of floats (allocated_amount values).
+		"""
+		bt = create_test_bank_transaction(self.bank_account, deposit=deposit, withdrawal=withdrawal)
+
+		if rows:
+			import random
+			import string
+
+			for amount in rows:
+				# Use a synthetic but unique child row name
+				row_name = "ABR-TEST-" + "".join(random.choices(string.ascii_lowercase, k=8))
```

Ruff (0.15.13)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)